packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -179,16 +191,50 @@ class CupertinoSwitch extends StatefulWidget {
   /// See also:
   ///
   ///  * [trackColor], the color to use for the track when the switch is off.
-  final Color? activeColor;
+  Color? get activeColor => _activeColor;
+
+  @Deprecated(
+      'Use activeTrackColor instead. '
+      'This feature was deprecated after v3.24.0-0.0.pre.'
+  )
+  final Color? _activeColor;
```

Comment:
You don't need a deprecation message on a private property. The deprecation messages on the constructor are enough. This is also the source of one of the test failures.